### PR TITLE
Implement getIterators for Spanner

### PR DESCRIPTION
**Pull request by gabrielfs7**
Implement getIterators for Spanner. PR also includes:

- PSR-12 compatibility.
- Test libraries updated
- Integration tests configurable by environment, not fixed in the code.
- Separation from integration tests from scripts.
- Some tests refactoring.

https://oat-sa.atlassian.net/browse/NEX-682

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
 /vendor/
 /.idea/
 /composer.lock
+/bin
+/tests/env.ini
```
Comment (mike-ionut-mihai-sandu-tao) [inline .gitignore]: i think we need a \n

Comment (gabrielfs7) [reply .gitignore]: done

File: composer.json [hunk 2]
```diff
@@ -14,8 +14,9 @@
         "phpmyadmin/sql-parser": "^5.2.0"
     },
     "require-dev":  {
-        "phpunit/phpunit": "^8.1",
-        "php-mock/php-mock": "^2.1"
+        "php-mock/php-mock": "^2.2",
+        "phpunit/phpunit": "^8.5",
+        "squizlabs/php_codesniffer": "^3.5"
```
Comment (mike-ionut-mihai-sandu-tao) [inline composer.json:19]: i don't think we need this , but it is also ok to leave it because it is required dev

File: src/SpannerPlatform.php [hunk 3]
```diff
@@ -164,7 +178,14 @@ public function getBlobTypeDeclarationSQL(array $field)
 
     public function getTruncateTableSQL($tableName, $cascade = false)
     {
-        throw new \Exception("\e[31m\e[1m" . __METHOD__ . "\e[21m\e[0m" . ' not yet implemented.' . "\n" . 'To implement it, please follow the guidelines here: https://stackoverflow.com/questions/43266590/does-cloud-spanner-support-a-truncate-table-command');
+        throw new Exception(
```
Comment (mike-ionut-mihai-sandu-tao) [inline src/SpannerPlatform.php:181]: maybe we can collect this into a status ticket

**Review by mike-ionut-mihai-sandu-tao — APPROVED** [review]
nice job

**Comment by mike-ionut-mihai-sandu-tao** [comment]
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Review by siwane — COMMENTED** [review]
Nice jobs, all cleanup is good and make easier to maintain. However, the iterator is aggregating the output but do not interfere in the statement itself. It is safe, but the original idea was to chunk iterator data. (iterate first 100 then query next 100 and append to iterator ...). It allows to control memory for big dataset in an transparent way.

wdyt

File: composer.json (hunk 2 above)
Comment (gabrielfs7) [reply composer.json:19]: IMO it is important to be here, so the library is self-contained with his dependencies and we ensure all devs are using the same version of codesniffer to format, etc.

**Comment by gabrielfs7** [comment]
> Nice jobs, all cleanup is good and make easier to maintain. However, the iterator is aggregating the output but do not interfere in the statement itself. It is safe, but the original idea was to chunk iterator data. (iterate first 100 then query next 100 and append to iterator ...). It allows to control memory for big dataset in an transparent way.
> 
> wdyt

I think we can use similar approach as Doctrine Paginator, cause the statement itself only executes a query, but does not change it. To do what you want, IMO the Statement is not the right place:

https://github.com/doctrine/orm/blob/master/lib/Doctrine/ORM/Tools/Pagination/Paginator.php

@mike-ionut-mihai-sandu-tao @boajer Please let here your opinion.

**Review by boajer — APPROVED** [review]
LGFM

- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Pull request merged**